app/models/setting.rb

```diff
@@ -49,7 +49,7 @@ def validate(record)
   validates :value, :format => { :with => Resolv::AddressRegex }, :if => proc { |s| IP_ATTRS.include? s.name }
   validates :value, :regexp => true, :if => proc { |s| REGEXP_ATTRS.include? s.name }
   validates :value, :array_type => true, :if => proc { |s| s.settings_type == "array" }
-  validates_with ValueValidator, :if => proc { |s| s.respond_to?("validate_#{s.name}") }
+  validates_with ValueValidator, if: proc { |s| s.respond_to?("validate_#{s.name}") }, on: :update
```

Comment:
Should the proc actually check if the definitions are loaded **and** it responds to the validate method?